File: app/src/main/java/pl/pola_app/helpers/CameraCompat.java
package pl.pola_app.helpers;


import android.graphics.Bitmap;
import android.view.SurfaceHolder;

public abstract class CameraCompat {

    public static CameraCompat create(int minPicSize) {
        return new OldCamera(minPicSize);
    }

    public abstract void open();

    public abstract void openPreview(SurfaceHolder surfaceholder);

    public abstract void closePreview();

    public abstract void takePicture(OnPhotoTakenSuccessListener onPhotoTakenSuccessListener);

    public abstract void release();

    public interface OnPhotoTakenSuccessListener {
        void onPhotoTakenSuccess(Bitmap bitmap, int originalWidth, int originalHeight, int width, int height);
    }
}

File: app/src/main/java/pl/pola_app/helpers/CameraCompat.kt
package pl.pola_app.helpers

import android.graphics.Bitmap
import android.view.SurfaceHolder

abstract class CameraCompat {
    abstract fun open()
    abstract fun openPreview(surfaceholder: SurfaceHolder)
    abstract fun closePreview()
    abstract fun takePicture(onPhotoTakenSuccessListener: OnPhotoTakenSuccessListener)
    abstract fun release()
    interface OnPhotoTakenSuccessListener {
        fun onPhotoTakenSuccess(
            bitmap: Bitmap,
            originalWidth: Int,
            originalHeight: Int,
            width: Int,
            height: Int
        )
    }

    companion object {
        fun create(minPicSize: Int): CameraCompat {
            return OldCamera(minPicSize)
        }
    }
}